File: data-jpa/src/test/java/top/bettercode/summer/data/jpa/domain/EmployeeKey.java
package top.bettercode.summer.data.jpa.domain;

import java.io.Serializable;
import java.util.Objects;
import javax.persistence.Embeddable;
import top.bettercode.summer.tools.lang.util.StringUtil;

/**
 * 客商档案 主键 对应表名：BD_CVDOC
 */
@Embeddable
public class EmployeeKey implements Serializable {

  private static final long serialVersionUID = 1L;

  private Integer id;
  private Integer id2;

  public EmployeeKey() {
  }

  public EmployeeKey(Integer id, Integer id2) {
    this.id = id;
    this.id2 = id2;
  }

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public Integer getId2() {
    return id2;
  }

  public void setId2(Integer id2) {
    this.id2 = id2;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof EmployeeKey)) {
      return false;
    }
    EmployeeKey that = (EmployeeKey) o;
    return Objects.equals(id, that.id) && Objects.equals(id2, that.id2);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id, id2);
  }

  @Override
  public String toString() {
    return StringUtil.json(this);
  }
}

File: data-jpa/src/test/kotlin/top/bettercode/summer/data/jpa/domain/EmployeeKey.kt
package top.bettercode.summer.data.jpa.domain

import top.bettercode.summer.tools.lang.util.StringUtil.json
import java.io.Serializable
import java.util.*
import javax.persistence.Embeddable

/**
 * 客商档案 主键 对应表名：BD_CVDOC
 */
@Embeddable
class EmployeeKey : Serializable {
    var id: Int? = null
    var id2: Int? = null

    constructor()
    constructor(id: Int?, id2: Int?) {
        this.id = id
        this.id2 = id2
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other !is EmployeeKey) {
            return false
        }
        return id == other.id && id2 == other.id2
    }

    override fun hashCode(): Int {
        return Objects.hash(id, id2)
    }

    override fun toString(): String {
        return json(this)
    }

}